hooks/useApiRequest.ts

```diff
@@ -0,0 +1,34 @@
+import { useSafeResponseSolve } from '@/hooks/useSafeResponseSolve';
+import { LoadingState } from '@/types/loading-state';
+import { AxiosResponse } from 'axios';
+import { useCallback, useEffect, useState } from 'react';
```

Comment:
```suggestion
import { AxiosResponse } from 'axios';
import { useCallback, useEffect, useState } from 'react';

import { useSafeResponseSolve } from '@/hooks/useSafeResponseSolve';
import { LoadingState } from '@/types/loading-state';
```

另外这个 LoadingState 不应该放在 `@/types` 目录下面

Reply:
> 另外这个 LoadingState 不应该放在 `@/types` 目录下面

放在这个hook文件里面？